Add katello_activation_key module

akofink opened the pull request:
Requires https://github.com/SatelliteQE/nailgun/pull/419

ehelms commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,201 @@
+        required: true
+    organization:
+        description:
+            - Organization label that the activation key is in
```
This should be Organization name and not label correct? Or should this support either?Or just label since its static?

ehelms commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,201 @@
+    organization: Default Organization
+    lifecycle_environment: "Library"
+    content_view:
+      - name: 'client content view'
```
This example does not meet the docs, I feel like it should just be `content_view: client content view'`

ehelms commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,201 @@
+        if len(response) == 1:
+            activation_key = response[0]
+        elif len(response) == 0:
+            activation_key = activation_key.create()
```
This only does a create and not an update. If the name of the activation key does not change, but I change the content view or lifecycle environment, I want that change to result in an 'update' to the activation key.

ehelms commented:
Thoughts on adding subscription support? That's one of the biggest reasons to make an Akey.

akofink commented:
Yeah, I'll go ahead and add subscription support :lollipop:

akofink commented:
Ready to go! Subscriptions and all. Requires https://github.com/SatelliteQE/nailgun/pull/419

evgeni commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,244 @@
+        description:
+            - Name of the content view
+        required: false
+'''
```
missing `subscriptions` option

evgeni commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,244 @@
+    password: "admin"
+    server_url: "https://foreman.example.com"
+    name: "Katello Clients"
+    organization: "\"Default Organization\""
```
why the double quoting?

evgeni commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,244 @@
+    lifecycle_environment: "Library"
+    content_view: 'client content view'
+    subscriptions:
+        - name: "\"Red Hat Enterprise Linux\""
```
why the double quoting?

evgeni commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,244 @@
+  delegate_to: localhost
+'''
+
+RETURN = '''# '''
```
what is this for?

akofink replied:
I honestly copied it from prior modules, but it's an [ansible module documentation directive](http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#return-block) showing what the module returns.

evgeni replied:
aaah, right! thanks.

evgeni commented on requirements.txt:
```diff
@@ -1 +1 @@
-nailgun
+git+https://github.com/SatelliteQE/nailgun.git@master#egg=nailgun
```
iek, should we just wait until a proper nailgun release with your changes is out?

akofink commented:
@evgeni I think I addressed all your comments. From speaking to QE (nailgun maintainers), I was recommended to use the master branch, rather than the latest release. I believe this is mostly because they don't release very often. If we could push them to release more regularly, then this wouldn't be an issue.

evgeni commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,248 @@
+
+    if not HAS_NAILGUN_PACKAGE:
+        module.fail_json(msg="Missing required nailgun module (check docs or install "
+                             "with: pip install nailgun")
```
missing closing parenthesis for "(check docs …" ;)

evgeni commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,248 @@
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
```
should mention the default is false. (there is a `default` key in the docs yaml)

evgeni commented on modules/katello_activation_key.py:
```diff
@@ -0,0 +1,248 @@
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(required=False, type='bool', default=False),
+            name=dict(required=True, no_log=False),
+            organization=dict(required=False, no_log=False),
```
shouldn't the org be required? otherwise it fails for me with:

```
FAILED! => {"changed": false, "failed": true, "msg": "No organization found for None"}
```

evgeni commented:
should be good to merge as soon the comments are adressed.

evgeni commented:
Oh, and I think in future we might want to add support for:
* Description
* Host Limit
* Service Level
* Release Version
* Host Collections

But I don't see that as a blocker.

akofink commented:
Updated

evgeni commented:
I think you have set the org as required in the doc string, but not in the module definition? ;)

akofink commented:
Whoops! Updated.
